plutus-ledger-constraints/src/Ledger/Constraints/OnChain/V1.hs

```diff
@@ -111,7 +111,7 @@ checkTxConstraint ctx@ScriptContext{scriptContextTxInfo} = \case
         -- gives us the redeemer's hash, but 'MustSpendScriptOutput' gives
         -- us the full redeemer
         $ isJust (V.findTxInByTxOutRef txOutRef scriptContextTxInfo)
-    MustMintValue mps _ tn v ->
+    MustMintValue mps _ tn v _ ->
```

Comment:
I think it's ok to assume it'll only get this far without a reference script in the tx. The same is done for MustSpendScriptOutput.

Reply:
I agree with @james-iohk that we don't have to check because when it is not `Nothing` construction of the transaction would have already failed.

Of course it is possible to construct the transaction yourself and use this constraint only in the script, and then a "cheating" transaction could store the minting script in the body instead of using the reference script. Perhaps this could theoretically be a security risk in some way? I think it's fine to just document that `MustMintValue` does not check on-chain that a reference script is used, instead of wasting valuable script resources to do the check.

Reply:
> store the minting script in the body instead of using the reference script

This wouldn't pass phase-1 because reference inputs are prohibited when V1 scripts are involved.
Even if using V2, you cannot have the same script in the txbody and as reference.